packages/plugin-ext/src/main/node/paths/plugin-paths-service.ts

```diff
@@ -95,7 +99,13 @@ export class PluginPathsServiceImpl implements PluginPathsService {
      * Generate time folder name in format: YYYYMMDDTHHMMSS, for example: 20181205T093828
      */
     private gererateTimeFolderName(): string {
-        return new Date().toISOString().replace(/[-:]|(\..*)/g, '');
+        const timeStamp = new Date().toISOString().replace(/[-:]|(\..*)/g, '');
+        // Ensures our timestamp generation logic is "valid".
+        // Changes to the timestamp structure may break old logs deletion logic.
+        if (!SESSION_TIMESTAMP_PATTERN.test(timeStamp)) {
+            throw new Error(`Failed Generating Session TimeStamp: <${timeStamp}>`);
```

Comment:
The old implementation would simply return a `string`, but now clients can potentially be thrown an error.
Should we update clients to perform error handling?

Reply:
or don't throw, just log an error?

Reply:
The intent here was not to expend the methods API to include `@throws`
Rather it was to ensure that the **created** folderNames are **aligned** with the **deleted** folder names. as a mis-alignments will break the cleanup feature...

This is sort of like a compiler check at runtime... I want to "break everything"
If somebody breaks the alignment, however I am not sure a "throws" will suffice.
So nobody can **merge** changes to master that break this alignment.

So what should I do here?
- Some hacky `process.exit(666)` to ensure breakage?
- Is there some other way to halt Theia with a meaningful error message?
- Logging and hoping somebody will look at the logs? (won't prevent merging code that causes mis-alignment).

Thanks.
Shahar.

Reply:
logging should be fine IMHO

Reply:
I've changed it to log an error